packages/core/lib/commands/test/run.js

```diff
@@ -30,6 +32,27 @@ const parseCommandLineFlags = options => {
   }
 };
 
+// Sanitize ganache options specified in the config
+function sanitizeGanacheOptions(ganacheOptions) {
+  let network_id = ganacheOptions.network_id;
+
+  // Use default network_id if "*" is defined in config
+  if (network_id === "*") {
+    network_id = defaultNetworkIdForTestCommand;
+    return { ...ganacheOptions, network_id };
+  }
+
+  const parsedNetworkId = parseInt(network_id, 10);
+  if (isNaN(parsedNetworkId)) {
+    const error =
+      `The network id specified in the truffle config ` +
+      `(${network_id}) is not valid. Please properly configure the network id as an integer value.`;
+    throw new Error(error);
+  }
+  network_id = parsedNetworkId;
+  return { ...ganacheOptions, network_id };
```

Comment:
This can be simplified
```suggestion
  return { ...ganacheOptions, network_id: parsedNetworkId };
```